crates/l2/proposer/l1_watcher.rs

```diff
@@ -69,6 +44,46 @@ impl L1Watcher {
             max_block_step: watcher_config.max_block_step,
             last_block_fetched: U256::zero(),
             l2_proposer_pk: watcher_config.l2_proposer_private_key,
+            check_interval: Duration::from_millis(watcher_config.check_interval_ms),
+        }
+    }
+
+    pub async fn run(&mut self, store: &Store) {
+        loop {
+            if let Err(err) = self.main_logic(store.clone()).await {
+                error!("L1 Watcher Error: {}", err);
+            }
+
+            sleep(Duration::from_millis(200)).await;
+        }
+    }
+
+    async fn main_logic(&mut self, store: Store) -> Result<(), L1WatcherError> {
+        loop {
+            sleep(self.check_interval).await;
+
+            let logs = match self.get_logs().await {
+                Ok(logs) => logs,
+                Err(error) => {
+                    warn!("Error when getting logs from L1: {}", error);
+                    continue;
+                }
+            };
+            if logs.is_empty() {
+                continue;
+            }
+
+            let pending_deposits_logs = match self.get_pending_deposit_logs().await {
+                Ok(logs) => logs,
+                Err(error) => {
+                    warn!("Error when getting L1 pending deposit logs: {}", error);
```

Comment:
Same here, we just bubble up this error above

Reply:
fixed